client/init.c

```diff
@@ -205,75 +207,74 @@ TDNFRefreshSack(
         pTdnf->pArgs->nRefresh = 1;
     }
 
-    dwError = TDNFInitCmdLineRepo(pTdnf, pSack);
-    BAIL_ON_TDNF_ERROR(dwError);
-
-    /* First repo is the "@cmdline" repo, which always exists.
-     * skip over it - options do not apply, and it is initialized. */
-    if(pTdnf->pRepos->pNext)
+    for(pRepo = pTdnf->pRepos; pRepo; pRepo = pRepo->pNext)
     {
-        PTDNF_REPO_DATA_INTERNAL pTempRepo = pTdnf->pRepos->pNext;
-        while(pTempRepo)
+        /* skip the @cmdline repo - options do not apply, and it is
+           initialized. */
+        if (strcmp(pRepo->pszName, "@cmdline") == 0)
+        {
+            continue;
+        }
+
+        if(!pRepo->nEnabled)
         {
-            nMetadataExpired = 0;
-            if(pTempRepo->nEnabled)
+            continue;
+        }
+
+        nMetadataExpired = 0;
+        //Check if expired since last sync per metadata_expire
+        if(pRepo->lMetadataExpire >= 0)
+        {
+            dwError = TDNFAllocateStringPrintf(
+                          &pszRepoCacheDir,
+                          "%s/%s",
+                          pTdnf->pConf->pszCacheDir,
+                          pRepo->pszId);
+            BAIL_ON_TDNF_ERROR(dwError);
+
+            dwError = TDNFShouldSyncMetadata(
+                          pszRepoCacheDir,
+                          pRepo->lMetadataExpire,
+                          &nMetadataExpired);
+            BAIL_ON_TDNF_ERROR(dwError);
+
+            TDNF_SAFE_FREE_MEMORY(pszRepoCacheDir);
+            pszRepoCacheDir = NULL;
+        }
+
+        if(nMetadataExpired)
+        {
+            dwError = TDNFRepoRemoveCache(pTdnf, pRepo->pszId);
```

Comment:
Should we also remove rpmcache based on pTdnf->pConf->nKeepCache check?

Reply:
I think not. `TDNFRepoRemoveCache)` removes the metadata only. I think the `keepcache` option is meant just for packages, which are not removed here.